### ISSUE TITLE
torch.compile(dynamic=True) fails on F.cross_entropy with probability targets and class weights

### ISSUE THREAD

**Issue by cangtianhuang** [pytorch/pytorch#182399]
### 🐛 Describe the bug

`torch.nn.functional.cross_entropy` works in eager mode when probability targets and per-class weights are used together, but the same function fails after being compiled with `torch.compile(dynamic=True)`.

This appears to be an eager/compiled execution-path inconsistency. The eager path returns a valid loss tensor, while the compiled path fails during fake tensor / symbolic shape tracing. The failure is triggered by the coexistence of probability targets and `weight` under dynamic-shape compilation.

Minimal reproducer:

```python
import torch
import torch.nn.functional as F

def fn(logits, prob_targets, weight):
    return F.cross_entropy(
        logits,
        prob_targets,
        weight=weight,
        reduction="none",
    )

logits = torch.randn(4, 3)
prob_targets = torch.rand(4, 3)
prob_targets = prob_targets / prob_targets.sum(dim=1, keepdim=True)
weight = torch.tensor([1.0, 2.0, 3.0])

# Eager path succeeds.
out_eager = fn(logits, prob_targets, weight)
print("eager:", out_eager.shape, out_eager.dtype)

# Compiled path fails.
compiled_fn = torch.compile(fn, dynamic=True)
out_compiled = compiled_fn(logits, prob_targets, weight)
print("compiled:", out_compiled.shape, out_compiled.dtype)
````

Observed behavior: eager execution succeeds and returns a loss tensor, but the compiled path raises an error during fake tensor / symbolic shape dispatch.

Expected behavior: `torch.compile(dynamic=True)` should either produce a result equivalent to eager execution, or raise a semantically consistent and user-facing error before entering an internal symbolic-shape failure.

### Versions

Collecting environment information...
PyTorch version: 2.6.0+cu126
Is debug build: False
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 家庭版 中文版 (10.0.26200 64 位)
GCC version: Could not collect
Clang version: 20.1.0
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.13 (main, Mar  3 2026, 15:01:35) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: True
CUDA runtime version: 12.6.20
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4060 Laptop GPU
Nvidia driver version: 595.97
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\bin\cudnn_ops64_9.dll
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Name: AMD Ryzen 7 7745HX with Radeon Graphics        
Manufacturer: AuthenticAMD
Family: 107
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3601
MaxClockSpeed: 3601
L2CacheSize: 8192
L2CacheSpeed: None
Revision: 24834

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @nkhasbag-nv @chauhang @penguinwu @ezyang @bobrenjc93 @aditvenk @laithsakka
